tools/misc/tpm2_eventlog.c

```diff
@@ -27,10 +31,33 @@ static bool on_positional(int argc, char **argv) {
     return true;
 }
 
+static bool on_option(char key, char *value) {
+
+    uint32_t version;
+
+    switch (key) {
+    case 'y':
+        version = (uint32_t)atoi(value);
+        for (unsigned int i = 0; i < sizeof(allowed_eventlog_versions)/sizeof(int); i++) {
+            if (version == allowed_eventlog_versions[i]) {
+                eventlog_version = version;
+                return true;
+            }
+        }
```

Comment:
ETOOCOMPLICATED, just set two defines in tpm2_eventlog_yaml.h:
```C
#define MIN_EVLOG_YAML_VERSION 1
#define MAX_EVLOG_YAML_VERSION 2
```
Then here do:
```C
if (version < MIN_EVLOG_YAML_VERSION || version > MAX_EVLOG_YAML_VERSION) {
  FAIL
}
```

We also probably want that logic in the functions that take YAML version.

Reply:
ok, will do